users/ignoring.rst

```diff
@@ -88,6 +88,32 @@ The ``.stignore`` file contains a list of file or path patterns. The
    are *included* (that is, *not* ignored). This can be used to override
    more general patterns that follow.
 
+   .. note::
+
+      Negated patterns that can match items below the folder root will cause
+      Syncthing to traverse otherwise ignored directories. If the
+      :ref:`watcher <scanning>` is enabled, those directories will also be
+      watched. Directories ignored before the first negated pattern can
+      however be safely skipped, since the first matching pattern wins. For
+      example::
+
+         /foo
+         /bar
+         !baz
+         *
+
+      The directories ``foo`` and ``bar`` will be entirely ignored. However any
```

Comment:
A note about the version where this optimized behavior was introduced would help.  Yes, we do have the docs versioning, but an unsuspecting user might not come up with the idea that the handling was ever different.